javascript/src/refactor.js

```diff
@@ -478,11 +476,39 @@ class RefactorEngine {
         return assignments;
     }
 
+    adjustFunctionParams(parameterList) {
+        var engine = this;
+        var assignments = {};
+        estraverse.replace(this.ast, {
+            enter: function (node, parent) {
+                if (node.type === 'Identifier' && parent.type === 'FunctionDeclaration') {
+                    for (let i = 0; i < parent.params.length; i++) {
+                        if (node.name === parent.params[i].name) {
+                            for (let j = 0; j < parameterList.length; j++) {
+                                if (
+                                    parameterList[j].functionName === parent.id.name &&
```

Comment:
Can we move this check outside of this for loop ? From what i understand, you are checking if the `parent` function declaration name is same as the function name of `parameter_list` right?  Maybe this could be like the first line inside `enter` ?

Reply:
Also do we also need to check that `parameterList.length == parent.params.length` ?

Reply:
@ketkarameya ```parameterList``` contains the details of the parameters to be removed across all the functions in the file. On the other hand, ```parent.params``` denotes only the arguments of a specific function declaration. Hence ```parameterList.length``` need not be same as ```parent.params.length```

Reply:
Then how do we know that `parent` is the correct method declaration? Will this handle overriden methods?

Reply:
> Can we move this check outside of this for loop ? From what i understand, you are checking if the `parent` function declaration name is same as the function name of `parameter_list` right? Maybe this could be like the first line inside `enter` ?

```parameterList``` is a list of objects, each of which has ```functionName, parameterIndex, value```. So even if its brought under ```enter```, we need to iterate of the list of objects available in ```parameterList```

Reply:
> 

got it.

Reply:
> Then how do we know that `parent` is the correct method declaration? Will this handle overriden methods?

In order to handle overriden functions, I've used your clue : checking ```parameterList[j].length == parent.params.length```
I'll update this in the new commit and let you know